src/types/frozenset.rs

```diff
@@ -8,6 +8,41 @@ use crate::{ffi, AsPyPointer, PyAny, Python, ToPyObject};
 
 use std::ptr;
 
+/// Allows building a Python `frozenset` one item at a time
+pub struct PyFrozenSetBuilder {
+    py_frozen_set: Py<PyFrozenSet>,
```

Comment:
I wonder if these builders would ever be long-lived, i.e. maybe making them GIL-bound as in

```rust
pub struct PyFrozenSetBuilder<'py>(&'py PyFrozenSet);
```

would be more ergonomic as it avoids passing a GIL token to `add` and `finalize`?